proposals/init.md

```diff
@@ -0,0 +1,490 @@
+compelling factors were (in no particular order):
+
+1. Desire to discourage older compilers from violating `init` semantics
+1. Desire to make adding or removing `init` in a `virtual` declaratio or 
+`interface` both a source and binary breaking change.
+
+Given there was also no significant support for removing `init` to be a 
+binary compatible change it made the choice of using modreq straight forward.
+
+### init vs. initonly
+Syntax debate time.
+
+**Resolution**
+There was no other syntax which was overwelming favored in LDM. For the time 
+being we will be moving forward with the initial design laid out in this 
+document. This may change as the feature progresses and LDM members have time
+to digest the design here.
+
+### Warn on failed init
+Consider the following scenario. A type declares an `init` only member which
+is not set in the constructor. Should the code which constructs the object 
+get a warning if they failed to initialize the value?
+
+At that point it is clear the field will never be set and hence has a lot of
+similarities with the warning around failing to initialize `private` data. 
+Hence a warning would seemingly have some value here?
+
+There are significant downsides to this warning though:
+1. It complicates the compatibility story of changing `readonly` to `init`. 
+1. It requires carrying additional metadata around to denote the members
+which are required to be initialized by the caller.
+
+Further if we believe there is value here in the overall scenario of forcing
+object creators to be warned / error'd about specific fields then this 
+likely makes sense as a general feature. There is no reason it should be 
+limited to just `init` members.
+
+**Resolution**
+There will be no warning on consumption of `init` fields and properties.
+
+LDM wants to have a broader discussion on the idea of required fields and
+properties. That may cause us to come back and reconsider our position on
+`init` members and validation.
+
+### Allow init as a type modifier
+In the same way the `readonly` modifier can be applied to a `struct` to 
+automatically declare all fields as `readonly`, the `init` only modifier can
+be declared on a `struct` or `class` to automatically mark all fields as `init`.
+This means the following two type declarations are equivalent:
+
+```cs
+struct Point
+{
+    public init int X;
+    public init int Y;
+}
+
+// vs. 
+
+init struct Point
+{
+    public int X;
+    public int Y;
+}
+```
+
+**Resolution** 
+This feature is too *cute* here and conflicts with the `readonly struct` 
+feature on which it is based. The `readonly struct` feature is simple in that 
+it applies `readonly` to all members: fields, methods, etc ... The 
+`init struct` feature would only apply to fields. This actually ends up making
+it confusing for users. 
+
+Given that `init` is only valid on certain aspects of a type we rejected the 
+idea of having it as a type modifier.
+
+## Allow init as a field modifier
```

Comment:
Consider moving this before the previous section, given that the decision was based on `init fields` not being a thing.